edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/edgedeployment/DiffCollectionExtensions.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    public static class DiffCollectionExtensions
+    {
+        public static Diff<T> Diff<T>(this IEnumerable<T> desired, IEnumerable<T> existing, Func<T, string> selector)
+        {
+            var either = new Set<T>(desired.ToDictionary(selector));
```

Comment:
It's a minor thing, but can we keep the name consistent with "desired" and "existing"?  "new" and "old" maybe, or desiredSet and existingSet?